webv2/primary.go

```diff
@@ -0,0 +1,340 @@
+
+	for i := 0; i < len(pkeyrequest.Columns); i++ {
+
+		for j := 0; j < len(spannertable.Pks); j++ {
+
+			id := getColumnId(spannertable, spannertable.Pks[j].Col)
+
+			if pkeyrequest.Columns[i].ColumnId == id {
+
+				spannertable.Pks[j].Desc = pkeyrequest.Columns[i].Desc
+				spannertable.Pks[j].Order = pkeyrequest.Columns[i].Order
+			}
+
+		}
+	}
+}
+
+//addPrimaryKey insert primary key into list of IndexKey
+func addPrimaryKey(add []int, pkeyrequest PrimaryKeyRequest, spannertable ddl.CreateTable) []ddl.IndexKey {
+
+	pklist := []ddl.IndexKey{}
+
+	for _, val := range add {
+
+		for i := 0; i < len(pkeyrequest.Columns); i++ {
+
+			if val == pkeyrequest.Columns[i].ColumnId {
+
+				pkey := ddl.IndexKey{}
+				pkey.Col = getColumnName(spannertable, pkeyrequest.Columns[i].ColumnId)
+				pkey.Desc = pkeyrequest.Columns[i].Desc
+				pkey.Order = pkeyrequest.Columns[i].Order
+
+				pklist = append(pklist, pkey)
+			}
+		}
+	}
+	return pklist
+}
+
+//removePrimaryKey removed primary key from list of IndexKey
+func removePrimaryKey(remove []int, spannertable ddl.CreateTable) []ddl.IndexKey {
+
+	pklist := []ddl.IndexKey{}
+
+	for _, val := range remove {
+
+		colname := getColumnName(spannertable, val)
+
+		for i := 0; i < len(spannertable.Pks); i++ {
+
+			if spannertable.Pks[i].Col == colname {
+
+				pklist = append(spannertable.Pks[:i], spannertable.Pks[i+1:]...)
+			}
+		}
+	}
+	return pklist
+}
+
+//prepareResponse prepare response for primary key api
+func prepareResponse(pkeyrequest PrimaryKeyRequest, spannertable ddl.CreateTable) PrimaryKeyResponse {
+
+	var pKeyResponse PrimaryKeyResponse
+
+	pKeyResponse.TableId = pkeyrequest.TableId
+	pKeyResponse.PrimaryKeyId = pkeyrequest.PrimaryKeyId
+
+	var issyntheticpkey bool
+
+	//todo check with team
+	for i := 0; i < len(spannertable.ColNames); i++ {
+
+		if spannertable.ColNames[i] == "synth_id" {
+			issyntheticpkey = true
+		}
+	}
+
+	pKeyResponse.Synth = issyntheticpkey
+
+	for _, indexkey := range spannertable.Pks {
+
+		responsecolumn := Column{}
+
+		id := getColumnId(spannertable, indexkey.Col)
+		responsecolumn.ColumnId = id
+		responsecolumn.ColName = indexkey.Col
+		responsecolumn.Desc = indexkey.Desc
+		responsecolumn.Order = indexkey.Order
+
+		pKeyResponse.Columns = append(pKeyResponse.Columns, responsecolumn)
+	}
+	return pKeyResponse
+}
+
+//preparenewpklist prepare first list for difference
+func preparenewpklist(pkeyrequest PrimaryKeyRequest) []int {
+
+	newlist := []int{}
+
+	for i := 0; i < len(pkeyrequest.Columns); i++ {
+		newlist = append(newlist, pkeyrequest.Columns[i].ColumnId)
+	}
+	return newlist
+}
+
+//prepareoldpklist prepare second list for difference
+func prepareoldpklist(spannertable ddl.CreateTable) []int {
+
+	oldlist := []int{}
+
+	for i := 0; i < len(spannertable.Pks); i++ {
+		cid := getColumnId(spannertable, spannertable.Pks[i].Col)
+		oldlist = append(oldlist, cid)
+	}
+	return oldlist
+}
+
+func preparecolumnlist(spannertable ddl.CreateTable) []int {
```

Comment:
Update the function name and correct the case. This returns a list of Column Ids.